src/community/graticule/src/main/java/org/geotools/data/graticule/gridsupport/GridUtilities.java

```diff
@@ -0,0 +1,349 @@
+package org.geotools.data.graticule.gridsupport;
+
+import java.io.IOException;
+import java.text.Format;
+import java.util.Arrays;
+import java.util.List;
+import org.geotools.api.data.FeatureSource;
+import org.geotools.api.data.SimpleFeatureSource;
+import org.geotools.api.feature.simple.SimpleFeature;
+import org.geotools.api.feature.simple.SimpleFeatureType;
+import org.geotools.api.feature.type.FeatureType;
+import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
+import org.geotools.api.referencing.cs.CoordinateSystem;
+import org.geotools.api.referencing.cs.CoordinateSystemAxis;
+import org.geotools.api.style.Style;
+import org.geotools.data.graticule.gridsupport.XWilkinson.Label;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.grid.DefaultGridFeatureBuilder;
+import org.geotools.grid.Lines;
+import org.geotools.grid.oblong.Oblongs;
+import org.geotools.grid.ortholine.LineOrientation;
+import org.geotools.grid.ortholine.OrthoLineDef;
+import org.geotools.map.FeatureLayer;
+import org.geotools.map.Layer;
+import org.geotools.referencing.CRS;
+import org.geotools.referencing.CRS.AxisOrder;
+import org.geotools.referencing.CoordinateFormat;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.LineString;
+
+public class GridUtilities {
+
+    public static final double DELTA = 0.00001;
+
+    private GridUtilities() {
+        // stop people instantiating the class
+    }
+
+    /**
+     * Utility class to create a grid layer that neatly wraps the features provided.
+     *
+     * @param style - the style to draw the grid with
+     * @param gridBounds - the bounds to include with in the grid
+     * @return - a Layer that contains the grid.
+     * @throws IOException if something goes wrong
+     */
+    public static Layer createNiceGridLayer(Style style, ReferencedEnvelope gridBounds)
```

Comment:
This code does not seem to be used... in fact I don't see usage of the whole utility class, NiceScale, or XWilkinson, in the store at all? I'd limit code to what we can use and test.